hpccm/primitives/shell.py

```diff
@@ -50,17 +57,20 @@ def __str__(self):
                 return ' && \\\n'.join(s)
             elif hpccm.config.g_ctype == container_type.SINGULARITY:
                 # Format:
-                # %post
+                # %post [OR %appinstall app_name]
                 #     cmd1
                 #     cmd2
                 #     cmd3
-                s = ['%post']
-
-                # For consistency with Docker. Docker resets the
-                # working directory to '/' at the beginning of each
-                # 'RUN' instruction.
-                if self.chdir:
-                    s.append('    cd /')
+                if self._app:
+                    s = ['%appinstall {0}'.format(self._app)]
+                else:
+                    s = ['%post']
+                    # For consistency with Docker. Docker resets the
+                    # working directory to '/' at the beginning of each
+                    # 'RUN' instruction. Singularity %appinstall is already
+                    # run in its own working directory.
+                    if self.chdir:
```

Comment:
un-indent?

Reply:
No, `%appinstall` sets the current working directory to the app base dir (`/scif/apps/[name]`). This behavior should be preserved in my opinion.

Reply:
Make sense.  Can you please add a comment saying this?